### PR TITLE
Integrated MockAPI within the map logic

### PR DESCRIPTION
**Pull request by Colividiu**
The coordinates are now fetched and updated using http methods. The drones' movements are still faked by timed increments, however at a lower speed (1 sec), in order to prevent errors concerning too many http requests being sent

### CODE REVIEW

File: src/components/MapAP.tsx [hunk 1]
```diff
@@ -1,45 +1,55 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
 import { APIProvider, Map, Marker } from '@vis.gl/react-google-maps';
 import '../models/Drone'
 import '../styles/Map.css'
 import logo from '../assets/LogoCC.png';
 import { Drone } from '../models/Drone';
+import axios from 'axios';
```
Comment (tomoiagahoratiu) [inline src/components/MapAP.tsx:7]: you need to add the dependency for axios

File: src/components/MapAP.tsx [hunk 2]
```diff
@@ -1,45 +1,55 @@
...
-]);
+  const [DronesList, setDronesList] = useState<Drone[]>([]);
+  const API_BASE_URL = "https://66d998da4ad2f6b8ed5550ff.mockapi.io";
+
+  const fetchDrones = async () => {
+    try {
+      const response = await axios.get(`${API_BASE_URL}/Drone`);
+      setDronesList(response.data);
+    } catch (error) {
+      console.error("Error fetching drones data:", error);
```
Comment (tomoiagahoratiu) [inline src/components/MapAP.tsx]: we should remove all the console logs, and maybe create a scenario for the failing call

**Comment by Colividiu** [comment]
I made the changes we discussed and did the next task

**Review by tomoiagahoratiu — APPROVED** [review]

**Comment by tomoiagahoratiu** [comment]
good job 👍

**Pull request merged**